src/main/java/fr/insalyon/citi/golo/runtime/FunctionCallSupport.java

```diff
@@ -69,8 +73,13 @@ public static Object samFilter(Class<?> type, Object value) {
     return value;
   }
 
-  public static CallSite bootstrap(Lookup caller, String name, MethodType type, int constant) throws IllegalAccessException, ClassNotFoundException {
-    FunctionCallSite callSite = new FunctionCallSite(caller, name.replaceAll("#", "\\."), type, constant == 1);
+  public static CallSite bootstrap(Lookup caller, String name, MethodType type, Object... bsmArgs) throws IllegalAccessException, ClassNotFoundException {
+    boolean constant = ((int)bsmArgs[0]) == 1;
+    String[] argumentNames = new String[bsmArgs.length - 1];
+    for (int i = 0; i < bsmArgs.length -1; i++) {
```

Comment:
I have to redo my tests but the original array is an instance of `Object[]` and the target one is `String[]`. I'm sure I've tested it and it failed in the casting.
With the loop I'm sure to cast every elements.

I'll check again.